lib/config-seed/nginx.conf

```diff
@@ -0,0 +1,70 @@
+user  nginx;
+worker_processes  1;
+
+error_log  /dev/stderr notice;
+
+events {
+    worker_connections  1024;
+    use epoll;
+}
+
+http {
+    resolver 127.0.0.11 ipv6=off valid=30s;
+
+    include       mime.types;
+    default_type  application/octet-stream;
+
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+    access_log  /dev/stdout  main;
+
+    sendfile        on;
+    tcp_nopush      on;
+    tcp_nodelay     on;
+
+    keepalive_timeout  65;
+
+    gzip  on;
+
+
+
+    server {
+        listen 18443 ssl;
+        server_name    _; # Catch all, see http://nginx.org/en/docs/http/server_names.html
+
+        server_name overleaf-toolkit.com;
+        ssl_certificate      /certs/nginx_certificate.pem;
+        ssl_certificate_key  /certs/nginx_key.pem;
+
+        ssl_protocols               TLSv1 TLSv1.1 TLSv1.2;
+        ssl_prefer_server_ciphers   on;
+
+        # used cloudflares ciphers https://github.com/cloudflare/sslconfig/blob/master/conf
+        ssl_ciphers                 EECDH+CHACHA20:EECDH+AES128:RSA+AES128:EECDH+AES256:RSA+AES256:EECDH+3DES:RSA+3DES:!MD5;
+
+        # config to enable HSTS(HTTP Strict Transport Security) https://developer.mozilla.org/en-US/docs/Security/HTTP_Strict_Transport_Security
+        # to avoid ssl stripping https://en.wikipedia.org/wiki/SSL_stripping#SSL_stripping	
+        add_header Strict-Transport-Security "max-age=31536000; includeSubdomains;";
+
+        server_tokens off;
+
+        add_header X-Frame-Options SAMEORIGIN;
+
+        add_header X-Content-Type-Options nosniff;
```

Comment:
It looks to me like `web` adds these headers (among others) itself via helmet ([here](https://github.com/overleaf/web/blob/737e4519b2dfe314dee9d49d634bc4e303127980/app/src/infrastructure/Server.js#L216-L223)).

Reply:
Removing these from the config and restarting gives me:
```
curl -k --head https://localhost
HTTP/1.1 302 Found
Server: nginx
Date: Mon, 26 Apr 2021 10:45:43 GMT
Content-Type: text/plain; charset=utf-8
Content-Length: 28
Connection: keep-alive
Content-Language: en
X-Frame-Options: SAMEORIGIN
X-Download-Options: noopen
X-Content-Type-Options: nosniff
Referrer-Policy: origin-when-cross-origin
X-XSS-Protection: 1; mode=block
Location: /login
Vary: Accept
Set-Cookie: sharelatex.sid=s%3A6aPZ3AW3NqVinDIy5TQ5ltn1NBfyX9fF.zVLPZt6Hu%2F1cX9g0zjm6jK8yZF5oPrv2z%2B%2B0Cn04bJU; Path=/; Expires=Sat, 01 May 2021 10:45:43 GMT; HttpOnly; SameSite=Lax
Strict-Transport-Security: max-age=31536000; includeSubdomains;
```
So yes, there doesn't appear to be a need to add them in nginx.conf.

Reply:
Removed from `nginx.conf`